source/docs/example-projects/halloween-treasure-hunt.rst

```diff
@@ -0,0 +1,251 @@
+argument. We feel the hype, so we'll go with a ``- narrate "yay"``.
+
+Our script should be the following at this point:
+
+.. code-block:: dscript
+    :name: figure_1_2
+    :linenos:
+    :emphasize-lines: 5,6
+
+    Halloween_Treasure_Hunt:
+    type: world
+    debug: true
+    events:
+      on player left clicks pumpkin in Hub:
+      - narrate "yay"
+
+.. rst-class:: figurecaption
+
+**Figure 1.2** Our world script with a specific event
+
+It's time to *save* the script file, *reload* scripts ingame with ``/denizen reload scripts`` and *trigger the event* 
+by left clicking a pumpkin block in our Hub world. We should now be able to see a cute little ``yay`` in chat, along 
+with some debug information in the console, just as we expected. That's great, but we also have to *test* and make 
+sure the event is not being triggered when clicking other types of blocks, when right clicking, or when clicking in 
+another world.
+
+Core Functionality
+==================
+
+We're ready to move further ahead and actually give a *reward* to the player clicking the block. Since we're nice 
+server owners, the prize will be a free *diamond*. This is where the :guilabel:`give` command comes in handy. Its 
+*syntax* specifies a single required argument: ``[money/xp/<item>|...]``. In our case, what we want to give the player 
+is a diamond item, so we can ignore the *money* and *xp* options.
+
+.. note::
+    When reading command documentation, It's important to keep in mind that anything inside ``< >`` is *not literal* 
+    and needs to be replaced. Arguments enclosed in ``[ ]`` are *required*, while ``( )`` means an argument is 
+    *optional*. The ``/`` symbol separates argument *options*, meaning you have to choose one of them.
+
+Let's go ahead and specify ``diamond`` as the first argument of our give command. We don't have to worry about who to 
+give the diamond to, as the command will target the linked player by default. That is, the player that triggered the 
+event. The full command line will then be ``- give diamond``.
+
+Now it's time to make sure it works. After *saving* and *reloading* scripts again, it should be giving us a *diamond* 
+every time we click the *pumpkin*. While players will totally love this, we should probably avoid giving out unlimited 
+diamonds. That's easy to fix though, we just have to *remove* the pumpkin once it's clicked. If we do it before even 
+giving out the reward, we'll make sure it won't be clicked twice. 
+
+For this, we'll use the :guilabel:`modifyblock` command, which lets us specify a *location* and a *material*. Now we 
+only need to know which location was clicked by the player. Time to make use of *context* tags! This kind of tags are 
+event specific and will let us retrieve useful information from said event. If we check again the event's 
+documentation, we can see it has a ``<context.location>`` tag available, which is just what we needed for the first 
```

Comment:
The idea of context tags will be covered in the world scripts section, and the specifics will be covered in the advanced world script topic in Section 4, so there's no need to re-explain what context tags are. You can just link back to the two sections (once they're written) as a "Hey, you should read these if you haven't already" reminder.